src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -82,79 +91,116 @@ case class WorkflowDescriptor(id: WorkflowId,
   lazy val writeToCache = configCallCaching && optionCacheWriting
   lazy val readFromCache = configCallCaching && optionCacheReading
 
-  lazy val workflowLogger = props.get("LOG_MODE") match {
-    case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
-      Paths.get(props.getOrElse("LOG_ROOT", ".")),
-      s"workflow.$id.log",
-      Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
-    )
-    case _ => NOPLogger.NOP_LOGGER
+  lazy val workflowLogName = s"workflow.$id.log"
+  lazy val workflowLogPath = props.get("LOG_MODE") match {
+    case Some(x) if x.toUpperCase.contains("SERVER") =>
+      Option(File(props.getOrElse("LOG_ROOT", ".")) / workflowLogName) map (_.path)
+    case _ => None
+  }
+  lazy val workflowLogger = workflowLogPath match {
+    case Some(path) => makeFileLogger(path, workflowLogName, Level.toLevel(props.getOrElse("LOG_LEVEL", "debug")))
+    case None => NOPLogger.NOP_LOGGER
   }
 
   lazy val workflowRootPath = wfContext.root.toPath(gcsFilesystem)
-  def workflowRootPathWithBaseRoot(rootPath: String): Path =
+  def workflowRootPathWithBaseRoot(rootPath: String): Path = {
     WorkflowDescriptor.buildWorkflowRootPath(rootPath, name, id).toPath(gcsFilesystem)
+  }
 
-  def copyWorkflowOutputs(implicit executionContext: ExecutionContext): Future[Unit] = {
+  def copyWorkflowOutputs(workflowMetadataResponse: WorkflowMetadataResponse)
```

Comment:
Realizing that the reason the stuff is here is the stuff was already here but I was already pretty concerned about how much stuff WorkflowDescriptor has accumulated over time. What about moving all of the copying logic out elsewhere?